File: bindings/kotlin/src/test/kotlin/com/ariacompute/engine/SetupTest.kt
package com.ariacompute.engine

import java.io.File
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith
import kotlin.test.assertFalse

class SetupTest {
    @Test
    fun defaults() {
        val cfg = defaultSetupConfig()
        assertEquals("auto", cfg.compute)
        assertEquals("", cfg.router)
    }

    @Test
    fun invalidEnum() {
        assertFailsWith<IllegalArgumentException> {
            applySetupFields(defaultSetupConfig(), compute = "gpu")
        }
    }

    @Test
    fun fillUrlsFromCnSite() {
        val got = fillSetupUrls(SetupConfig(siteUrl = CN_SITE))
        assertEquals(CN_UPGRADE, got.upgradeUrl)
    }

    @Test
    fun allFields() {
        val st = applySetupFields(
            defaultSetupConfig(),
            router = "http://127.0.0.1:8080",
            siteUrl = CN_SITE,
            upgradeUrl = CN_UPGRADE,
            compute = "cpu",
            hfToken = "hf_abc",
            modelscopeApiToken = "ms_xyz",
        )
        assertEquals("http://127.0.0.1:8080", st.router)
        assertEquals("cpu", st.compute)
        assertEquals("hf_abc", st.hfToken)
    }

    @Test
    fun doesNotWriteConfigYml() {
        val home = File.createTempFile("aria-kt", "home")
        home.delete()
        home.mkdirs()
        System.setProperty("user.home", home.absolutePath)
        applySetupFields(defaultSetupConfig(), router = "http://127.0.0.1:8080", hfToken = "hf_x")
        assertFalse(File(home, "config.yml").isFile)
    }
}

File: bindings/flutter/test/setup_test.dart
import 'package:ariaengine/setup.dart';
import 'package:test/test.dart';

void main() {
  test('defaults', () {
    final cfg = defaultSetupConfig();
    expect(cfg.compute, 'auto');
    expect(cfg.router, '');
  });

  test('invalid enum', () {
    expect(
        () => applySetup(defaultSetupConfig(), compute: 'gpu'), throwsArgumentError);
  });

  test('fill urls from cn site', () {
    final got = fillSetupUrls(SetupConfig(siteUrl: cnSite));
    expect(got.upgradeUrl, cnUpgrade);
  });

  test('all fields roundtrip', () {
    final st = applySetup(
      defaultSetupConfig(),
      router: 'http://127.0.0.1:8080',
      siteUrl: cnSite,
      upgradeUrl: cnUpgrade,
      compute: 'cpu',
      hfToken: 'hf_abc',
      modelscopeApiToken: 'ms_xyz',
    );
    expect(st.router, 'http://127.0.0.1:8080');
    expect(st.compute, 'cpu');
    expect(st.hfToken, 'hf_abc');
  });

  test('invalid enum leaves state', () {
    final once = applySetup(defaultSetupConfig(), compute: 'cpu');
    expect(() => applySetup(once, compute: 'gpu'), throwsArgumentError);
    expect(once.compute, 'cpu');
  });
}
